features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/AsyncApiConsole/AsyncApiConsole.jsx

```diff
@@ -0,0 +1,256 @@
+                        return endpoint.environmentName;
+                    });
+                    setEnvironments(environment);
+                }
+                if (apiData.labels) {
+                    const Label = apiData.labels.map((label) => {
+                        return label.name;
+                    });
+                    setLabels(Label);
+                }
+                if (apiData.scopes) {
+                    const scopeList = apiData.scopes.map((scope) => {
+                        return scope.name;
+                    });
+                    setScopes(scopeList);
+                }
+            })
+            .catch((error) => {
+                if (process.env.NODE_ENV !== 'production') {
+                    console.error(error);
+                }
+                const { status } = error;
+                if (status === 404) {
+                    setNotFound(true);
+                }
+            });
+    }, []);
+
+    /**
+     * Load the access token for given key type
+     */
+    function updateAccessToken() {
+        let accessToken;
+        if (keys.get(selectedKeyType)) {
+            ({ accessToken } = keys.get(selectedKeyType).token);
+        }
+        if (selectedKeyType === 'PRODUCTION') {
+            setProductionAccessToken(accessToken);
+        } else {
+            setSandboxAccessToken(accessToken);
+        }
+    }
+
+    /**
+     * set Password
+     * @param {*} selectedKey
+     * @param {*} isUpdateToken
+     */
+    function setSelectedKeyType(selectedKey, isUpdateToken) {
+        if (isUpdateToken) {
+            setSelectedKey(selectedKey, updateAccessToken);
+        } else {
+            setSelectedKey(selectedKey);
+        }
+    }
+
+
+    function accessTokenProvider() {
+        if (securitySchemeType === 'BASIC') {
+            const credentials = username + ':' + password;
+            return btoa(credentials);
+        }
+        if (securitySchemeType === 'API-KEY') {
+            if (selectedKeyType === 'PRODUCTION') {
+                return productionApiKey;
+            } else {
+                return sandboxApiKey;
+            }
+        } else if (selectedKeyType === 'PRODUCTION') {
+            return productionAccessToken;
+        } else {
+            return sandboxAccessToken;
+        }
+    }
+
+    if (api == null) {
+        return <Progress />;
+    }
+    if (notFound) {
+        return 'API Not found !';
+    }
+
+    let isApiKeyEnabled = false;
+
+    let authorizationHeader = api.authorizationHeader ? api.authorizationHeader : 'Authorization';
+
+    if (api && api.securityScheme) {
+        isApiKeyEnabled = api.securityScheme.includes('api_key');
+        if (isApiKeyEnabled && securitySchemeType === 'API-KEY') {
+            authorizationHeader = 'apikey';
+        }
+    }
+
+    const isPrototypedAPI = api.lifeCycleStatus && api.lifeCycleStatus.toLowerCase() === 'prototyped';
+    return (
```

Comment:
Shall we set the initial state of scopeList, Label, environment, to null and show loading while the response comes. 
```javascript
if(!scopeList || label || environment) {
return ( <Progress /> );
}
```

Reply:
else part was handled in the catch as discussed